html/cross-origin-embedder-policy/require-corp-about-srcdoc.html

```diff
@@ -4,30 +4,28 @@
 
 promise_test(t => {
   return new Promise(resolve => {
-    window.addEventListener("DOMContentLoaded", resolve);
+    window.addEventListener("DOMContentLoaded", t.step_func(resolve));
   });
 }, "Wait for the DOM to be built.");
 
 promise_test(async t => {
   let iframe = document.createElement("iframe");
-  let iframe_loaded =  new Promise(resolve => iframe.onload = resolve);
+  let iframe_loaded =  new Promise(resolve => iframe.onload = t.step_func(resolve));
   iframe.srcdoc = "loaded document";
   document.body.appendChild(iframe);
 
   // The about:srcdoc document can load.
   await iframe_loaded;
   assert_not_equals(iframe.contentDocument, null);
   assert_equals(iframe.contentDocument.body.innerText, "loaded document");
-
-  t.done();
 }, "about:srcdoc can always be embedded by a 'require-corp' document");
 
 promise_test(async t => {
   let iframe_C = document.createElement("iframe");
   let iframe_B = document.createElement("iframe");
   iframe_B.srcdoc = "dummy content";
   iframe_C.src = "/common/blank.html";
-  let iframe_B_loaded = new Promise(resolve => iframe_B.onload = resolve);
+  let iframe_B_loaded = new Promise(resolve => iframe_B.onload = t.step_func(resolve));
```

Comment:
Is that really necessary? my understanding is it would be if "resolve" was a function that'd do some assertions/throws : `Note that in the promise chain constructed in test_function assertions don’t need to be wrapped in step or step_func calls. However when mixing event handlers and promise_test, the event handler callback functions do need to be wrapped since an exception in these functions does not cause the promise chain to reject.`

Reply:
That's a good question. I guess it's not, but I'm worried that people might not realize that when modifying this. @jgraham what do you think?

Reply:
I think you're both right :) Since bare resolve can't throw, it doesn't need to be wrapped, and wrapping it adds a little boilerplate. But it's also true that any user function does have to be wrapped and so consistently wrapping everything might avoid bugs in the future.

Reply:
I believe I did the mistake on other promise tests, so leaving them as a "reminder" is ok with me.

Reply:
FWIW, I realized I am inconsistent here: https://github.com/web-platform-tests/wpt/commit/9dc1d43944ceac2f8ca2d8d12fc9ccd7da38c7b2.